### PR TITLE
tech: refactor github comment to use Capgemini Microsoft Bot

### PR DESCRIPTION
**Pull request by tdashworth**
## Purpose
This PR replaces the out-of-the-box GitHub task to create a comment with a custom task that uses a GitHub App to make the request. This is preferred so that it's not a named user. 

## Approach
As above.

## TODOs
- [ ] Automated test coverage for new code
- [ ] Documentation updated (if required)
- [ ] Build and tests successful

### CODE REVIEW

**Comment by tdashworth** [comment]
Please note, that the build (test) failures are not related to these changes. These are being investigated separately.

**Comment by tdashworth** [comment]
Closes #126

**Review by osagiestar — APPROVED** [review]
Great idea and nice to have

File: templates/build-and-test-stages.yml [hunk 1]
```diff
@@ -141,4 +144,11 @@ stages:
             continueOnError: true
             inputs:
               authenticationType: PowerPlatformSPN
-              PowerPlatformSPN: Dataverse (placeholder)
\ No newline at end of file
+              PowerPlatformSPN: Dataverse (placeholder)
+          - task: GitHubAppDeleteIssueComment@1
+            condition: ne(variables['CreateComment.CommentId'], '')
```
Comment (ewingjm) [inline templates/build-and-test-stages.yml:149]: Is it worth deleting the comments? Could see it being useful having a history.

**Review by ewingjm — APPROVED** [review]

**Pull request merged**